Userland/Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -336,6 +336,10 @@ Vector<MatchingRule> StyleComputer::collect_matching_rules(DOM::Element const& e
     }
     if (auto it = rule_cache.rules_by_tag_name.find(element.local_name()); it != rule_cache.rules_by_tag_name.end())
         add_rules_to_run(it->value);
+    if (pseudo_element.has_value())
+        add_rules_to_run(rule_cache.pseudo_element_rules);
+    if (is<HTML::HTMLHtmlElement>(element))
+        add_rules_to_run(rule_cache.root_rules);
```

Comment:
Is the root guaranteed to be an HTML element? As far as I'm aware it should match the root of an SVG document too, and maybe other things.